lib/Avatar/AvatarWithInformation.js

```diff
@@ -0,0 +1,87 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import { Avatar } from '../';
+
+class AvatarWithInformation extends Component {
+  constructor() {
+    super();
+    this.state = {
+      showAdditional: false
+    };
+  }
+
+  showAdditional = () => this.setState({ showAdditional: true });
+  hideAdditional = () => this.setState({ showAdditional: false });
+
+  render() {
+    const additionalClasses = cx(
+      `avatar__wrapper avatar__with-text ${this.props.className}`,
+      {
+        'avatar--offline': this.props.isOffline,
+        'avatar--additional-visible':
+          this.props.additional && this.state.showAdditional,
+        'avatar__with-text--size-sm': this.props.smallSize
+      }
+    );
+    return (
+      <div
+        className={additionalClasses}
+        onMouseEnter={this.showAdditional}
+        onMouseLeave={this.hideAdditional}
+      >
+        <Avatar
+          colour={this.props.colour}
+          name={this.props.name}
+          url={this.props.url}
+          initials={this.props.initials}
+          isOffline={this.props.isOffline}
+          isHighlighted={this.props.isHighlighted}
+          email={this.props.email}
+          smallSize={this.props.smallSize}
+        />
+        <div className="avatar__information">
+          {this.props.name && (
+            <span className="avatar__name">{this.props.name}</span>
+          )}
+
+          {this.props.email && (
+            <span className="avatar__email">{this.props.email}</span>
+          )}
+        </div>
+        {this.props.additional &&
+          this.state.showAdditional && (
```

Comment:
After adding my feedback about moving the information content into a child component I wondered if this is worth taking over to the Avatar component too?

If additional information was rendered but was visually hidden and shown via CSS...would that be bad? It would remove the need for more component logic...unless we need to remove the additional content from the DOM for a reason?

If so...how do screen readers see this content?